lib/features/login/presentation/screen/login_screen.dart

```diff
@@ -152,4 +153,14 @@ class _LoginScreenState extends State<LoginScreen> {
       }
     }
   }
+
+  Future<void> _continueToLoggedInContext() async {
+    final result = await _checkIfAppTermsShouldBeShownUseCase.checkIfAppTermsShouldBeShown();
+
+    if (result == true) {
+      context.router.replace(const AppTermsRoute());
+    } else {
+      context.router.replace(const HomeRoute());
+    }
```

Comment:
Please add "if (mounted)"

Reply:
Added

Reply:
I think you should guard both branches

Android Studio underlines some possible problems, and if you hover a mouse it says

> Don't use 'BuildContext's across async gaps. (Documentation)  
> Try rewriting the code to not use the 'BuildContext', or guard the use with a 'mounted' check.

<img width="405" alt="image" src="https://github.com/zachran-jidlo/zachranobed/assets/80771513/594f797a-6d81-412b-91f4-93a69f2b9eec">

As for me you can just make a "quick return" if not mounted:
```dart
if (!mounted) {
  return;
}
if (result == true) {
  context.router.replace(const AppTermsRoute());
} else {
  context.router.replace(const HomeRoute());
}
```

Reply:
Whoah, I must have been blind. Thank you for noticing